app/src/main/java/org/akvo/flow/presentation/form/FormPresenter.kt

```diff
@@ -124,4 +128,20 @@ class FormPresenter @Inject constructor(
             view?.dismiss()
         }
     }
+
+    fun updateInstanceVersion(readOnly: Boolean, form: @NotNull Survey, formInstanceId: Long) {
+        if (readOnly || formInstanceId == -1L) {
+            return
+        }
+        //update submission form version
+        uiScope.launch {
+            val params: MutableMap<String, Any> = HashMap(4)
+            params[UpdateFormInstance.PARAM_FORM_INSTANCE_ID] = formInstanceId
+            params[UpdateFormInstance.PARAM_FORM_VERSION] = form.version
+            val result: UpdateFormInstance.FormVersionUpdateResult = updateFormInstance.execute(params)
+            if (result == UpdateFormInstance.FormVersionUpdateResult.FormVersionUpdated) {
+                view?.showFormUpdated()
```

Comment:
Does the updated popup show for every form instance that is updated?  That could cause a weird UX issue if its the case.. no?

Reply:
ok I am now only showing the message once